From f1d77f973035950856cbdfe29c008742b43cee55 Mon Sep 17 00:00:00 2001
From: Jacksgong <igzhenjie@gmail.com>
Date: Mon, 6 Aug 2018 16:53:46 +0800
Subject: [PATCH 222/296] Fix the IndexOutOfBoundsException issue.

Check if the size is greater than 0
before mPendingInstalls.remove(0).
---
 .../core/java/com/android/server/pm/PackageManagerService.java    | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 19e7a92dfc1..b34f6374bef 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -1606,9 +1606,11 @@ public class PackageManagerService extends IPackageManager.Stub
                 }
                 case MCS_GIVE_UP: {
                     if (DEBUG_INSTALL) Slog.i(TAG, "mcs_giveup too many retries");
-                    HandlerParams params = mPendingInstalls.remove(0);
-                    Trace.asyncTraceEnd(TRACE_TAG_PACKAGE_MANAGER, "queueInstall",
-                            System.identityHashCode(params));
+                    if (mPendingInstalls.size() > 0) {
+                        HandlerParams params = mPendingInstalls.remove(0);
+                        Trace.asyncTraceEnd(TRACE_TAG_PACKAGE_MANAGER, "queueInstall",
+                                System.identityHashCode(params));
+                    }
                     break;
                 }
                 case SEND_PENDING_BROADCAST: {
-- 
2.11.0

